### PR TITLE
Settle order with paidAt

### PR DESCRIPTION
**Pull request by falklinus**
# Jira Issue: [CLX-247]

## What?
- Allow requesting multiple values from `confirmWithValue`
- Ask for paidAt on manual payments
- Refactor/move confirm logic to the `usePaymentOrder` hook

## Optional checklist
- [ ] Updated `src/changelog.ts`
- [x] Codescouted
- [ ] Unit tests written
- [x] Tested locally

[CLX-247]: https://hedvig.atlassian.net/browse/CLX-247?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: shared/@hedvig-ui/Modal/use-confirm-dialog.tsx [hunk 1]
```diff
@@ -67,31 +68,50 @@ export const ConfirmDialogComponent: React.FC<{
...
-                checked={checkBoxChecked}
-              />
-            ) : valueType === 'input' ? (
-              <Input placeholder={valueLabel} ref={valueRef} />
-            ) : null}
-          </Flex>
-        )}
+        {withValue &&
+          !!valueTypes &&
+          Object.entries(valueTypes).map(([name, { label, type }]) => (
```
Comment (vonElfvin) [inline shared/@hedvig-ui/Modal/use-confirm-dialog.tsx:73]: Smort

File: shared/@hedvig-ui/Modal/use-confirm-dialog.tsx [hunk 2]
```diff
@@ -43,11 +44,11 @@ export const ConfirmDialogComponent: React.FC<{
   visible,
   status = 'success',
   withValue = false,
-  valueType,
-  valueLabel,
+  valueTypes,
 }) => {
-  const valueRef = useRef<HTMLInputElement>(null)
-  const [checkBoxChecked, setCheckBoxChecked] = useState(false)
+  const [values, setValues] =
```
Comment (vonElfvin) [inline shared/@hedvig-ui/Modal/use-confirm-dialog.tsx:49]: This is some old school react state management 😄  I have a faint memory this is where `useReduce` should be used but I think pulling in yet another hook is a bit overkill here

File: src/portals/hope/features/paymentOrders/hooks/use-payment-orders.tsx [hunk 3]
```diff
@@ -222,25 +232,99 @@ export const usePaymentOrder = () => {
...
+        error: ({ message }) => extractErrorMessage(message),
+      }),
+    )
   }
-  const settlePaymentOrder = async (id: string) => {
-    await toast.promise(settleOrder({ variables: { id } }), {
-      loading: 'Settling',
-      success: 'Order settled',
-      error: ({ message }) => extractErrorMessage(message),
+  const settlePaymentOrder = async (order: PaymentOrderInformationFragment) => {
```
Comment (vonElfvin) [inline src/portals/hope/features/paymentOrders/hooks/use-payment-orders.tsx:250]: Nice DRY refactoring

**Review by vonElfvin — APPROVED** [review]
Nice "Don't live with broken windows" refactorings in here as well (https://en.wikipedia.org/wiki/Broken_windows_theory)

File: shared/@hedvig-ui/Modal/use-confirm-dialog.tsx (hunk 2 above)
Comment (falklinus) [reply shared/@hedvig-ui/Modal/use-confirm-dialog.tsx:49]: True, could probably do with some predefined updater functions, but this should be sufficient 🧑‍🍳

**Pull request merged**